pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -180,6 +180,11 @@ message SingleMessageMetadata {
     optional bool null_partition_key = 10 [ default = false];
 }
 
+// raw metadata for message
+message RawMessageMetadata {
```

Comment:
I think the original idea to introduce RawMessageMetadata is to avoid serialization and deserialization. So broker doesn't need to serialize another protobuf message and create a new entry to store in bookkeeper side. I am not sure why do we need this.

Reply:
> When the broker side gets the message produced from the client, we could add the property at a new area, which does not combine with MessageMetadata, and no need deserializing the original MessageMetadata when gets it out; and when the broker sends the message to the client, we could choose to filter out this part of the property(or not as the client needs). 

As described in [PIP-70](https://github.com/apache/pulsar/wiki/PIP-70%3A-Introduce-lightweight-raw-Message-metadata),  RawMessageMetadata just avoid serialization and deserialization for the original  MessageMetadata and can be  fiter out before send to client.

Here the protobuf message RawMessageMetadata is just a Tool Class for build/parse Raw meta.